pywbem/cim_operations.py

```diff
@@ -1172,12 +1237,302 @@ def __repr__(self):
+
+        Raises:
+
+          ValueError: Input contains an invalid log destination, log level,
+            or log detail level. No named logger is configured.
+        """
+        handler = cls._configure_logger_handler(log_dest, log_filename)
+
+        detail_level = cls._validate_detail_level(detail_level)
+
+        cls._configure_logger(LOG_HTTP_NAME, detail_level,
+                              handler, connection)
+
+    @classmethod
+    def configure_all_loggers(cls, log_dest=DEFAULT_LOG_DESTINATION,
+                              detail_level=None,
+                              log_filename=DEFAULT_LOG_DESTINATION,
+                              connection=None):
+        """
+        A log configureation method that sets both the api and loggers to
+        the parameters define in the call
+
+        Parameters:
+
+          log_dest (:term:`string`):
+            String defining the destination for this log. It must be one of the
+            destinations defined in :data:`~pywbem._logging.LOG_DESTINATIONS`
+            or `None`. If the value is the string ``none``, the null logger is
+            configured.
+
+          detail_level (:term:`string` or :class: `int`):
+            If :term:`string` it must be one of the key words that
+            defines the level of detail to be output. The possible
+            key words are 'all', 'summary', 'paths'.
+            TODO should get this list from variable.
+
+            If  :class: `int` it defines the maximum size of the log record
+            generated.
+
+          log_filename (:term:`string`):
+            Filename to use as logging file if the log destination is `file`.
+            Ignored if log destination is not `file`. If value is `None` and
+            this is a ``file`` log, ValueError is raised.
+
+          connection (:class:`~pywbem.WBEMConnection`)
+            Connection to activate for logging.
+
+            If `None` the no WBEMConnection is to be activated.
+
+            if :class:`py:bool` and `True` all subsequent connections are
+            activated for logging
+
+            If an object of :class:`~pywbem.WBEMConnection` logging of
+            the api calls and responses of the connectionactivated using the
+            log configuration provided in log_dest and detail_level
+            parameters or configured using standard python logging configuration
+            methods.
+
+        Raises:
+
+          ValueError: Input contains an invalid log destination, log level,
+            or log detail level. No named logger is configured.
+        """
+        cls.configure_api_logger(log_dest=log_dest, log_filename=log_filename,
+                                 detail_level=detail_level,
+                                 connection=connection)
+        cls.configure_http_logger(log_dest=log_dest, log_filename=log_filename,
+                                  detail_level=detail_level,
+                                  connection=connection)
```

Comment:
We now have three configuration methods with the same parameters that only differ in the logger(s) they control. Should we not make the logger name (or better some shorthand for it that we use in the CLI) a parameter, with "all" meaning both of them?

Reply:
DONE. We now have one method